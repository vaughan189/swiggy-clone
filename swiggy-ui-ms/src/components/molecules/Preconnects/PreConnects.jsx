/**
 *
 * PreConnects
 *
 */
import React from 'react';

const PreConnects = ({ urlList = [] }) =>
  urlList.length
    ? urlList.map((item) => {
        return <link rel="preconnect" href={item} key={item} />;
      })
    : null;

export default PreConnects;
