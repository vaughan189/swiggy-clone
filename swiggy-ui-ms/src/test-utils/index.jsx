import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { theme } from '../theme';

const AllTheProviders = ({ children, mocks = [] }) => {
  return (
    <MockedProvider mocks={mocks} addTypename={false}>
        {children}
    </MockedProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
  mocks: PropTypes.arrayOf(PropTypes.shape({})),
};

AllTheProviders.defaultProps = {
  mocks: [],
};

const customRender = (ui, options) => {
  const { wrapperProps = {}, ...customOptions } = options || {};
  return render(ui, {
    wrapper: (props) => {
      return <AllTheProviders {...props} {...wrapperProps} />;
    },
    ...customOptions,
  });
};

// re-export everything
export * from '@testing-library/react';
export * from './withTestRouter';

// override render method
export { customRender as render };
