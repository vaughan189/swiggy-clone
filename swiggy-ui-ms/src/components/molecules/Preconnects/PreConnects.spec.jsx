import React from 'react';
import { render } from '../../../test-utils';

import PreConnects from '.';

describe('PreConnects', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<PreConnects urlList={['/test']} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render null if urlList is not present', () => {
    const { asFragment } = render(<PreConnects />);
    expect(asFragment()).toMatchSnapshot();
  });
});
