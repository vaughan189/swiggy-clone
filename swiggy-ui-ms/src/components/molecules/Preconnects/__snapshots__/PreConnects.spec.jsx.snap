// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreConnects should render correctly 1`] = `
<DocumentFragment>
  <link
    href="/test"
    rel="preconnect"
  />
</DocumentFragment>
`;

exports[`PreConnects should render null if urlList is not present 1`] = `<DocumentFragment />`;
